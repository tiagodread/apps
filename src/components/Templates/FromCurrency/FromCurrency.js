import React from 'react';
import { Form } from 'react-bootstrap';
import PropTypes from 'prop-types';

function FromCurrency({
  fromCurrency,
  handleChangeFrom,
}) {

  return (
    <Form.Select aria-label="Select from currency" data-testid="from-currency" value={fromCurrency} onChange={handleChangeFrom}>
      <option value="EUR">EUR</option>
      <option value="USD">USD</option>
      <option value="GBP">GBP</option>
      <option value="JPY">JPY</option>
      <option value="BRL">BRL</option>
      <option value="CAD">CAD</option>
    </Form.Select>
  );
}

FromCurrency.propTypes = {
  fromCurrency: PropTypes.string.isRequired,
  handleChangeFrom: PropTypes.func.isRequired,
};

FromCurrency.defaultProps = {
  fromCurrency: 'USD',
  handleChangeFrom: () => { },
};

export default FromCurrency;