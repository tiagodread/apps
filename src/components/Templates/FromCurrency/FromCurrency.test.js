import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import FromCurrency from './FromCurrency';

describe('FromCurrency', () => {
    it('renders without crashing', () => {
        render(<FromCurrency />);
    });

    it('renders the correct initial currency state', () => {
        render(<FromCurrency fromCurrency="CAD"  />);
        const fromCurrency = screen.getByTestId('from-currency');
        expect(fromCurrency.value).toBe('CAD');
    });

    it('should change currency state when the user changes the from currency', () => {
        render(<FromCurrency fromCurrency="CAD" />);
        const fromCurrency = screen.getByTestId('from-currency');
        expect(fromCurrency.value).toBe('CAD');
        fromCurrency.value = 'USD';
        expect(fromCurrency.value).toBe('USD');
    });

    it('Should match the snapshot', () => {
        const { container } = render(<FromCurrency fromCurrency="CAD" />);
        expect(container).toMatchSnapshot();
    });
});
