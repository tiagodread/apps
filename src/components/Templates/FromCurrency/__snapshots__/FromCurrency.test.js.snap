// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FromCurrency Should match the snapshot 1`] = `
<div>
  <select
    aria-label="Select from currency"
    class="form-select"
    data-testid="from-currency"
  >
    <option
      value="EUR"
    >
      EUR
    </option>
    <option
      value="USD"
    >
      USD
    </option>
    <option
      value="GBP"
    >
      GBP
    </option>
    <option
      value="JPY"
    >
      JPY
    </option>
    <option
      value="BRL"
    >
      BRL
    </option>
    <option
      value="CAD"
    >
      CAD
    </option>
  </select>
</div>
`;
